docroot/profiles/custom/cgov_site/modules/custom/pdq_cancer_information_summary/src/Plugin/rest/resource/PDQResource.php

```diff
@@ -0,0 +1,271 @@
+    ];
+    foreach (['en', 'es'] as $code) {
+      if ($node->hasTranslation($code)) {
+        $translation = $node->getTranslation($code);
+        $sections = [];
+        foreach ($translation->field_summary_sections as $section) {
+          $s = Paragraph::load($section->target_id);
+          $sections[] = [
+            'id' => $s->field_pdq_section_id->value,
+            'title' => $s->field_pdq_section_title->value,
+            'html' => $s->field_pdq_section_html->value,
+          ];
+        }
+        $fields[$code] = [
+          'title' => $translation->getTitle(),
+          'cdr_id' => $translation->field_pdq_cdr_id->value,
+          'audience' => $translation->field_pdq_audience->value,
+          'summary_type' => $translation->field_pdq_summary_type->value,
+          'posted_date' => $translation->field_date_posted->value,
+          'updated_date' => $translation->field_date_updated->value,
+          'short_title' => $translation->field_short_title->value,
+          'description' => $translation->field_list_description->value,
+
+          // This field will probably not be retained for the intitial
+          // release.
+          // 'keywords' => $translation->field_syndication_keywords->value,
+          // End of suppressed field.
+          'public_use' => $translation->field_public_use->value,
+          'url' => $translation->path->alias,
+          'published' => $translation->status->value,
+          'sections' => $sections,
+        ];
+      }
+    }
+    $response = new ResourceResponse($fields);
+    $response->addCacheableDependency(['#cache' => ['max-age' => 0]]);
+    return $response;
+  }
+
+  /**
+   * Response to POST requests.
+   *
+   * Stores a new PDQ Summary node or update an existing node.
+   *
+   * Decided to use POST for replacing existing summary nodes,
+   * because PUT is required to be idempotent, meaning if the
+   * client submits the same request twice, the effect on the
+   * resource should be no different than if it had been
+   * submitted only once, and it would be prohibitively hard
+   * to figure out whether the document submitted each time
+   * were identical, and we should not create a new revision
+   * (we'll let the upstream client take care of that logic).
+   *
+   * See https://stackoverflow.com/questions/630453/put-vs-post-in-rest
+   *
+   * In contrast with the CDR, the Drupal CMS stores an English
+   * Cancer Information Summary document and it's Spanish translation
+   * document in the same node.
+   *
+   * @param array $summary
+   *   The summary document values.
+   *
+   * @return \Drupal\rest\ModifiedResourceResponse
+   *   The response containing node ID(s) or a Drupal node
+   *
+   * @throws \Symfony\Component\HttpKernel\Exception\HttpException
+   *   Throws exception expected.
+   */
+  public function post(array $summary) {
+
+    // Extract the bits we'll use frequently.
+    $nid = $summary['nid'];
```

Comment:
Could the service look up the NID based on the CDRID? Was this decision because a new CDRID could replace an existing summary?

Reply:
I mean you could have passed the English summary ID in with the spanish and then CDR would not have to know anything about Drupal... However, I can live with this for now.